go/frontendbridge/index.go

```diff
@@ -18,46 +18,48 @@ import (
 
 var upgrader = websocket.Upgrader{}
 
-// Connection represents a connection
-type Connection struct {
-	socket *websocket.Conn
-	auth   structs.MessageAuth
-}
-
 // FrontendBridge handles communication between an Exosphere backend and
 // a browser front-end using websockets
 type FrontendBridge struct {
 	connections      []*Connection
 	connectionsMutex sync.RWMutex
-	exoRelay         exorelay.ExoRelay
-	exoRelayMutex    sync.Mutex
+	exoSocket        *exosocket.ExoSocket
+	exoSocketMutex   sync.Mutex
 	open             bool
 	openMutex        sync.RWMutex
 	server           http.Server
 }
 
-// Open sets up connection to exocom and a server to listen for connections from clients
-func (w *FrontendBridge) Open(config exorelay.Config, clientPort int) error {
-	err := w.connectToExocom(config)
-	if err != nil {
-		return err
+// NewFrontendBridge returns a FrontendBridge instance
+func NewFrontendBridge(config exosocket.Config, clientPort int) *FrontendBridge {
+	w := &FrontendBridge{
+		exoSocket:   exosocket.NewExoSocket(config),
+		connections: []*Connection{},
 	}
-	err = w.listenForClients(clientPort)
-	if err != nil {
-		return err
+	var handler http.HandlerFunc = func(resWriter http.ResponseWriter, req *http.Request) {
```

Comment:
Thoughts on extracting a `WebsocketServer` class here, which encapsulates the `http.Server` class? Then the FrontendBridge doesn't need to be concerned with upgrading basic HTTP connections to websockets, and that functionality becomes reusable.